src/jsMain/kotlin/org/hisp/dhis/rules/RuleEngineJs.kt

```diff
@@ -38,9 +39,9 @@ class RuleEngineJs {
             .map(::toRuleEffectJs).toTypedArray()
     }
 
-    private fun <Kf, Vf, K, V> toMap(map: Array<Entry<Kf, Vf>>, key: (Kf) -> K, value: (Vf) -> V): Map<K, V> {
+    private fun <Kf, Vf, K, V> toMap(map: JsMap<Kf, Vf>, key: (Kf) -> K, value: (Vf) -> V): Map<K, V> {
         val res : MutableMap<K, V> = mutableMapOf()
-        map.forEach { e -> res[key(e.key)] = value(e.value) }
+        map.forEach { v, k -> res[key(k)] = value(v) }
```

Comment:
This looks backwards, I*d expect `k,v`, is this indeed defined like this?

Reply:
It is `v, k` because it is defined like that in JS Map class and the wrapper must mimic the syntax (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach#syntax).
Actually, if I change the order it doesn't compile, it is type-safe

Reply:
FYI it is defined in this class https://github.com/JetBrains/kotlin-wrappers/blob/master/kotlin-js/src/jsMain/kotlin/js/collections/JsMap.kt